src/dist.h

```diff
@@ -0,0 +1,15 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+/** @file dist.h
+    @brief Defines function with distribution specific implementations.
+*/
+
+/** Print the binary type: slim or whole. */
+void printBinaryType();
+
+/** Initializes distribution specific paths based on main executable abspath. */
+void initializeShareDirPath(const char *MainExecutableAbsPath);
```

Comment:
Since composite binary doesn't use share dir path I suggest to rename this method to just `initialize`.